src/models/fields/base-field-configuration.ts

```diff
@@ -1,36 +1,96 @@
-  inputType: Joi.string().valid('text', 'textarea', 'select', 'radio', 'switch', 'checkbox', 'file', 'files')
+  type: Joi.string().valid('string', 'boolean', 'number', 'integer')
+    .default('string')
+    .when(Joi.ref('regex'), {
+      is: Joi.string().required(),
+      then: Joi.string().valid('string').default('string'),
+    })
+    .when(Joi.ref('lowerbound'), {
+      is: Joi.number().required(),
+      then: Joi.string().valid('number').default('number'),
+    })
+    .when(Joi.ref('upperbound'), {
+      is: Joi.number().required(),
+      then: Joi.string().valid('number').default('number'),
+    }),
+  name: stringOrTranslationsSchema.required(),
+  inputType: Joi.string()
+    .when(Joi.ref('valueOptions'), {
+      is: Joi.array().required(),
+      then: Joi.string()
+        .valid('select')
+        .default('select'),
+      otherwise: Joi.string()
+        .valid('text', 'textarea', 'select', 'radio', 'switch', 'checkbox', 'file', 'files')
+        .default('text'),
+    })
     .description('The UI component to show. If not specified, it is text unless valueOptions are provided, then it is select.'),
+  defaultValue: Joi
+    .when(Joi.ref('type'), {
+      is: 'string',
+      then: Joi.string().default(''),
+    })
+    .when(Joi.ref('type'), {
+      is: 'number',
+      then: Joi.number().default(0),
+    })
+    .when(Joi.ref('type'), {
+      is: 'integer',
+      then: Joi.number().integer().default(0),
+    })
+    .when(Joi.ref('type'), {
+      is: 'boolean',
+      then: Joi.boolean().default(0),
+    })
+    .required(),
   valueOptions: Joi.array().items(Joi.object().keys({
-    text: Joi.string().required(),
+    text: stringOrTranslationsSchema.required(),
     value: baseFieldSchema.required().description('Will be passed through parser'),
-  })).allow(null)
-    .description('If null, inputType should not be select or radio. If not null, input type should be select or radio'),
-  prefix: Joi.string().description('Not available for inputTypes \'radio\', \'switch\', \'checkbox\', \'file\' and \'files\''),
-  suffix: Joi.string().description('Not available for inputTypes \'radio\', \'switch\', \'checkbox\', \'file\' and \'files\''),
+  })).allow(null).default(null)
+    .description('If null, inputType should not be select or radio. If not null, input type should be select or radio. Default: null'),
+  regex: Joi.string().description('If provided, type should be string and the provided value should adhere to this regex'),
+  lowerbound: Joi.number().description('If provided, type should be number or integer (and lowerbound should be integer as well in case type is integer) and provided value should not be lower than this value'),
+  upperbound: Joi.number().description('If provided, type should be number or integer (and lowerbound should be integer as well in case type is integer) and provided value should not be higher than this value'),
```

Comment:
Could you drop the `(and lowerbound should be integer as well in case type is integer)` part in the description? I know I wrote it myself, but I think it's clearer without.